### PR TITLE
Adds documentation regarding Ansible and Playbooks

### PR DESCRIPTION
**Pull request by xploitedd**

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,195 @@
+# Infrastructure Documentation
+
+## Why Ansible
+
+Currently there are several options for provisioning a VM such as Puppet or Chef but in the end Ansible was the winner.
```
Comment (pmhsfelix) [inline docs/README.md]: We will probably use some of this information in the final report. In that context, perhaps avoid sentences as `the winner`. E.g. replace with `the select option` or `the chosen option`.

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+  tasks: 
+    # ...
+    - name: Copy nginx config
+      copy:
+        src: "{{ nginx_file }}" # We can use the variables with Mustache templating
+        dest: /etc/nginx/sites-available/
+    - name: Enable nginx virtual host
+      file:
+        path: /etc/nginx/sites-enabled/
+        src: "/etc/nginx/sites-available/{{ nginx_file }}" # Here too :)
```
Comment (pmhsfelix) [inline docs/README.md:157]: Is it possible to apply templating to the copied file? For instance, I would like to replicate this process with a VM I created, but with a DNS name other than `iondev.live`. This means the content of the nginx file need to be changed.

**Review by pmhsfelix — APPROVED** [review]

File: docs/README.md (hunk 2 above)
Comment (xploitedd) [reply docs/README.md:157]: I think we could use https://docs.ansible.com/ansible/latest/collections/ansible/builtin/lineinfile_module.html for this purpose.

File: docs/README.md (hunk 1 above)
Comment (xploitedd) [reply docs/README.md]: Going to change that, thanks for the input!

**Review by pmhsfelix — APPROVED** [review]
Great docs. See some comments that I left.

File: docs/README.md [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+    - secrets.yml # This file has a varible "secret" defined
+  tasks: 
+    - name: Executes important command
+      shell: "important-command --our-secret={{ secret }}" # We use our secret that was imported from the variables file
+```
+
+Furthermore, Ansible allows us to use Vaults: we can encrypt our variable files, providing an extra layer of security (or the only layer of security if the file is in source control), however, this feature was not explored as of yet, because it was deemed unnecessary for the time being.
+
+### Replace lines in a file with RegExp
+
```
Comment (pmhsfelix) [inline docs/README.md:181]: Great addition.

**Review by pmhsfelix — APPROVED** [review]

**Pull request merged**